internal/engine/compiler/engine.go

```diff
@@ -717,13 +745,6 @@ func (ce *callEngine) Call(ctx context.Context, params ...uint64) (results []uin
 		}
 	}
 
-	tp := ce.initialFn.funcType
-
-	paramCount := len(params)
-	if tp.ParamNumInUint64 != paramCount {
```

Comment:
This checked is moved to the wrappers, since it's different for each one.
This is a behavior change: arguments are checked before module closure.
Is that a problem?